### PR TITLE
feat(desktop): add first-class Mistral Vibe coding agent

### LINKED ISSUE #5551: [feat] First-class Mistral Vibe coding agent support
#### Describe the feature

Add **Mistral Vibe** (`vibe`, Mistral's CLI coding agent) as a first-class terminal agent, on par with the existing Claude / Codex / Cursor / OpenCode / Gemini integrations — selectable in every agent picker, launchable into a workspace pane, with a model picker and working-indicator + completion-chime lifecycle.

#### Motivation

Vibe is a capable open coding agent, but today it can only be added as a generic custom agent (no icon, no model picker, no lifecycle hooks). First-class support lets users pick it like any other built-in agent and get the full Superset experience.

#### Proposed approach

Runs entirely through the existing terminal-agent architecture — **no changes to the Mistral Vibe CLI are required**:

- **Registry:** add `vibe` to `BUILTIN_TERMINAL_AGENTS` (`vibe --trust --auto-approve`); everything (types, labels, host presets, wire enums, pickers) derives from it. Positional prompt auto-submits into Vibe's TUI.
- **Icon:** wire a `vibe` preset icon + custom-agent icon option + default terminal preset.
- **Model picker:** Vibe has no `--model` flag, so add `modelEnv` support to `AgentModelSupport` + `buildAgentModelEnv` and inject `VIBE_ACTIVE_MODEL` into the launch env.
- **Lifecycle hooks:** a managed `vibe` PATH-wrapper (enables `VIBE_ENABLE_EXPERIMENTAL_HOOKS`) + a marker-guarded `~/.vibe/hooks.toml` calling the existing notify script; map `before_tool → Start` and `post_agent_turn → Stop`.
- **Docs/discovery:** settings search, AGENTS.md note, marketing/docs surfaces.

#### Out of scope (possible follow-ups)

- MCP wiring (Vibe uses TOML `[[mcp_servers]]`, not `.mcp.json`).
- ACP runtime (`vibe-acp`).
- Optional upstream Vibe hooks (turn-start / permission-requested) for finer working-state fidelity.

I have an implementation ready and will open a PR referencing this issue.

### PR DESCRIPTION
**Pull request by ThomsenDrake**
## Description

Adds **Mistral Vibe** (`vibe`) as a first-class terminal coding agent, matching the depth of the existing Claude / Codex integrations: selectable in every agent picker with its own icon, launches into a PTY pane with a seeded prompt, exposes a model picker (env-based `VIBE_ACTIVE_MODEL`, since Vibe has no `--model` flag), and drives the working-indicator + completion-chime lifecycle via a managed wrapper + a marker-guarded `~/.vibe/hooks.toml`. No changes to the Mistral Vibe CLI are required.

Everything flows from a single entry in `BUILTIN_TERMINAL_AGENTS`; the non-derived touch points (icon, model catalog, desktop hook pipeline, event mapping, docs) are updated to match.

## Related Issues

closes #5551

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation
- [ ] Refactor
- [ ] Other (please describe):

## Testing

- `bun run lint` clean (0 findings), `bun run typecheck` 28/28, `bunx sherif` clean, `bun install --frozen` no changes.
- New/updated suites pass 76/0: `agent-command`, `agent-models` (incl. `buildAgentModelEnv`), `map-event-type` (`before_tool → Start`, `post_agent_turn → Stop`), `agent-wrappers` (vibe wrapper + `~/.vibe/hooks.toml` writer).
- Manual smoke in a packaged desktop build: selected Mistral Vibe + a model, launched a task → pane runs `VIBE_ACTIVE_MODEL=<model> vibe --trust --auto-approve '<prompt>'`; working indicator lights on first tool call, completion chime fires once at idle. Verified the generated wrapper and `~/.vibe/hooks.toml` provisioning, and drove the full `superset` CLI lifecycle from inside a Vibe session.

## Additional Notes

- Model picker exposes Vibe's built-in aliases (`mistral-medium-3.5`, `devstral-small`); delivered via `VIBE_ACTIVE_MODEL` because Vibe has no `--model` flag.
- Out of scope (possible follow-ups): MCP wiring (Vibe uses TOML `[[mcp_servers]]`, not `.mcp.json`), the `vibe-acp` ACP runtime, and optional upstream Vibe hooks (turn-start / permission-requested) for finer working-state fidelity.

---

<a href="https://stagereview.app/superset-sh/superset/pull/5552">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://stagereview.app/assets/gh-open-in-stage-dark.svg">
    Open in Stage
  </picture>
</a>

---
## Summary by cubic
Adds Mistral Vibe (`vibe`) as a first-class terminal coding agent with full model selection and lifecycle hooks. Adds a shared notify-hook command and safer `hooks.toml` recovery to protect user-defined hooks.

- **New Features**
  - Registry/UI: added to `BUILTIN_TERMINAL_AGENTS`, default terminal presets, icon sets, settings search, docs/marketing/mobile.
  - Launch: runs `vibe --trust --auto-approve '<prompt>'` (positional prompt auto-submits).
  - Model picker: env-based via `VIBE_ACTIVE_MODEL`; added `modelEnv` and `buildAgentModelEnv`, merged into the host launch env.
  - Lifecycle: wrapper enables `VIBE_ENABLE_EXPERIMENTAL_HOOKS` and stamps `SUPERSET_AGENT_ID`; provisions marker-guarded `~/.vibe/hooks.toml` with `before_tool` and `post_agent_turn` hooks; mapped to `Start`/`Stop`.
  - Compatibility: Vibe reads `AGENTS.md` and `.agents/skills/` (with `--trust`); MCP uses TOML `[[mcp_servers]]` in `.vibe/config.toml`.
  - Robustness: `hooks.toml` recovery now preserves user hooks/comments when the end marker is missing by removing only managed `superset-notify-*` tables and stopping at the first foreign TOML table.
  - Refactor: extracted shared `getManagedNotifyHookCommand`/`MANAGED_NOTIFY_RELATIVE_PATH` in `agent-wrappers-common`; applied to Claude and Vibe.

- **Migration**
  - Ensure `vibe` is installed and `MISTRAL_API_KEY` is configured (or run `vibe --setup`).
  - No data migrations; the app auto-creates the wrapper and `~/.vibe/hooks.toml`. Model picker includes `mistral-medium-3.5` and `devstral-small`.

<sup>Written for commit f786a22ffd9f2b22fe998d51e46f3329c010127f. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/superset-sh/superset/pull/5552?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added **Mistral Vibe** (`vibe`) as a built-in agent across the app, including icons, marketing/UX surfaces, default terminal presets, and quick-add templates.
  * Enabled Vibe lifecycle hooks with managed startup setup and mapped “start/stop” event behavior in the UI.
  * Introduced env-based model selection for Vibe so model choice no longer relies on a CLI `--model` flag.
* **Bug Fixes**
  * Improved Vibe hook `hooks.toml` generation to be idempotent and robust against partial writes.
* **Documentation**
  * Updated compatibility notes and MCP configuration docs for Vibe.
* **Tests**
  * Added unit coverage for Vibe wrapper script, `hooks.toml` generation, and hook event mapping.

### CODE REVIEW

**Comment by ghost** [comment]
Ready to review this PR? Stage has broken it down into 7 individual chapters for you:

| | Title |
|---|-------|
| 1 | [Document the Mistral Vibe integration plan](https://stagereview.app/superset-sh/superset/pull/5552/chapters/1) |
| 2 | [Register Vibe as a builtin terminal agent](https://stagereview.app/superset-sh/superset/pull/5552/chapters/2) |
| 3 | [Support env-based model selection for Vibe](https://stagereview.app/superset-sh/superset/pull/5552/chapters/3) |
| 4 | [Implement lifecycle hooks and event mapping](https://stagereview.app/superset-sh/superset/pull/5552/chapters/4) |
| 5 | [Provision Vibe PATH-wrapper and hooks.toml](https://stagereview.app/superset-sh/superset/pull/5552/chapters/5) |
| 6 | [Wire Vibe into UI and documentation](https://stagereview.app/superset-sh/superset/pull/5552/chapters/6) |
| 7 | [Other changes](https://stagereview.app/superset-sh/superset/pull/5552/chapters/7) |

<a href="https://stagereview.app/superset-sh/superset/pull/5552">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://stagereview.app/assets/gh-open-in-stage-dark.svg">
    Open in Stage
  </picture>
</a>

<sub>Chapters generated by <a href="https://stagereview.app">Stage</a> for commit f786a22ffd9f2b22fe998d51e46f3329c010127f on Jul 9, 2026 10:55pm UTC.</sub>

File: apps/desktop/src/main/lib/agent-setup/agent-wrappers-vibe.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+export function getVibeHooksTomlContent(existing: string): string {
+	let base = existing;
+	const start = base.indexOf(VIBE_HOOKS_MARKER_START);
+	if (start !== -1) {
+		const end = base.indexOf(VIBE_HOOKS_MARKER_END, start);
+		if (end !== -1) {
+			base =
+				base.slice(0, start) + base.slice(end + VIBE_HOOKS_MARKER_END.length);
+		}
+	}
```
[the remark this answers is not in the document]
Comment (ThomsenDrake) [reply apps/desktop/src/main/lib/agent-setup/agent-wrappers-vibe.ts:92]: Good catch — fixed in 420b6cfaa.

`getVibeHooksTomlContent` now strips from the start marker to end-of-file when `VIBE_HOOKS_MARKER_END` is absent:

```ts
} else {
  // Orphaned start marker (partial/interrupted write) — strip to EOF.
  base = base.slice(0, start);
}
```

so an orphaned/partial block can no longer produce duplicate `[[hooks]]` entries or a dangling marker. Added a regression test ("cleans up an orphaned start marker left by a partial write") covering the missing-end-marker path.

File: apps/desktop/src/main/lib/agent-setup/agent-wrappers-vibe.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	base = base.replace(/\s+$/, "");
+	const block = buildVibeManagedHooksBlock();
+	return base.length > 0 ? `${base}\n\n${block}\n` : `${block}\n`;
+}
```
[the remark this answers is not in the document]
Comment (ThomsenDrake) [reply apps/desktop/src/main/lib/agent-setup/agent-wrappers-vibe.ts:96]: Good catch — fixed in 420b6cfaa.

`getVibeHooksTomlContent` now strips from the start marker to end-of-file when `VIBE_HOOKS_MARKER_END` is absent:

```ts
} else {
  // Orphaned start marker (partial/interrupted write) — strip to EOF.
  base = base.slice(0, start);
}
```

so an orphaned/partial block can no longer produce duplicate `[[hooks]]` entries or a dangling marker. Added a regression test ("cleans up an orphaned start marker left by a partial write") covering the missing-end-marker path.

File: apps/desktop/src/main/lib/agent-setup/agent-wrappers-vibe.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ * Merge our managed block into an existing hooks.toml: strip any prior managed
+ * block (between markers), then append the fresh one. Preserves user hooks and
+ * is idempotent — no TOML parser needed since we own the block content.
+ */
+export function getVibeHooksTomlContent(existing: string): string {
+	let base = existing;
+	const start = base.indexOf(VIBE_HOOKS_MARKER_START);
+	if (start !== -1) {
+		const end = base.indexOf(VIBE_HOOKS_MARKER_END, start);
+		if (end !== -1) {
```
[the remark this answers is not in the document]
Comment (ThomsenDrake) [reply apps/desktop/src/main/lib/agent-setup/agent-wrappers-vibe.ts]: Good catch — fixed in 420b6cfaa.

`getVibeHooksTomlContent` now strips from the start marker to end-of-file when `VIBE_HOOKS_MARKER_END` is absent:

```ts
} else {
  // Orphaned start marker (partial/interrupted write) — strip to EOF.
  base = base.slice(0, start);
}
```

so an orphaned/partial block can no longer produce duplicate `[[hooks]]` entries or a dangling marker. Added a regression test ("cleans up an orphaned start marker left by a partial write") covering the missing-end-marker path.

File: plans/20260709-mistral-vibe-agent-impl-plan.md [hunk 4]
```diff
@@ -0,0 +1,687 @@
...
+- [ ] **Step 2:** Commit: `git commit -am "docs(agents): note Mistral Vibe config/skills conventions"`.
+
+### Task 4.3 (optional): Marketing + docs surfaces
+
+**Files:** `apps/marketing/src/app/components/FeaturesSection/components/UniversalCompatibilityDemo/UniversalCompatibilityDemo.tsx`, `apps/marketing/src/app/components/HeroSection/components/AppMockup/constants.ts` (+ `apps/marketing/public/app-icons/vibe.svg`), `apps/docs/content/docs/mcp.mdx` + `terminal-presets.mdx`, `apps/mobile/store.config.json`.
+
+- [ ] **Step 1:** Add Mistral Vibe to the marketing agent lists (with a `vibe.svg` in `apps/marketing/public/app-icons/`), the docs agent/preset/MCP enumerations, and the mobile store keywords. Mirror existing entries.
+- [ ] **Step 2:** `bun run lint && bun run typecheck` → PASS.
+- [ ] **Step 3:** Commit: `git commit -am "docs(marketing): include Mistral Vibe in compatibility surfaces"`.
```
[the remark this answers is not in the document]
Comment (ThomsenDrake) [reply plans/20260709-mistral-vibe-agent-impl-plan.md:664]: These surfaces are all implemented in this PR — settings search, `AGENTS.md`, the marketing components (`UniversalCompatibilityDemo`, `AppMockup/constants`), the docs enumerations (`mcp.mdx`, `terminal-presets.mdx`), and the mobile store keywords are all included, so nothing ships without them. The "(optional)" label was planning-phase phrasing about trimmability, not a sign they were skipped. Happy to drop that wording from the plan doc if you'd prefer it read as required.

**Comment by ThomsenDrake** [comment]
Thanks for the reviews. Addressed the one actionable code finding — the orphaned start-marker edge case in the Vibe `hooks.toml` writer — in 420b6cfaa (fix + regression test).

On the CodeRabbit pre-merge warnings:

- **Out-of-scope changes:** the mobile store keywords and marketing/docs entries are the discovery surfaces that make the agent findable (the same treatment the existing built-in agents get), and the two `plans/` docs follow this repo's `plans/` convention (AGENTS.md rule #6). Happy to split any of these into a follow-up if a tighter diff is preferred.
- **Docstring coverage:** the new public helpers carry JSDoc; can expand coverage on the changed files if the threshold is enforced.

**Comment by ThomsenDrake** [comment]
Hey @Kitenite I'd love to get a review on this. Myself and a couple of my coworkers at Mistral love Superset and would love to be able to use our Vibe CLI on it!

**Comment by Kitenite** [comment]
Looks great! one last note while i was testing was auth broke if i needed to sign in first into browser. i had to sign in first and then rerun the auth to work

https://console.mistral.ai/codestral/cli/authenticate?process_id=bd0b560b-41b2-4372-96e6-f1fac172fa33&complete_token=n2Rx7J1PMnDPSGzML03GpPqnFCiJGL67

Screenshot 2026-07-09 at 3 36 46 PM

**Comment by Kitenite** [comment]
There's also no running hook only stop hook?

**Comment by Kitenite** [comment]
Note on the orphaned-marker fix (420b6cfaa): the strip-to-EOF (`base.slice(0, start)` when the end marker is missing) that all three bots suggested silently deletes any user `[[hooks]]` sitting below the managed block — it trades a transient duplicate for real data loss. Safer to strip only our own `superset-notify-*` tables and stop at the first foreign TOML table, so trailing user hooks survive. Verified: with a user hook after an end-marker-less block, strip-to-EOF drops it.
